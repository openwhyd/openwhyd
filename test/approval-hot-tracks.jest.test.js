// Run with: $ npm run test-approval-hot-tracks

const {
  httpClient,
  connectToMongoDB,
  startOpenwhydServer,
} = require('./approval-tests-helpers');

const { START_WITH_ENV_FILE } = process.env;

const MONGODB_URL =
  process.env.MONGODB_URL || 'mongodb://localhost:27117/openwhyd_test';

describe('Hot Tracks (approval tests - to be replaced later by unit tests)', () => {
  let mongoClient;
  let db;
  let server;

  beforeAll(async () => {
    // if this test times out, make sure to start MongoDB first: $ docker-compose up -d mongo
    mongoClient = await connectToMongoDB(MONGODB_URL);
    db = await mongoClient.db();
  });

  afterAll(async () => {
    await mongoClient.close();
    server?.kill('SIGKILL');
  });

  beforeEach(async () => {
    server?.kill('SIGKILL');
    await db.collection('user').deleteMany({});
    await db.collection('track').deleteMany({});
  });

  it('renders ranked tracks', async () => {
    await db.collection('track').insertMany([
      { name: 'a regular track', score: 1 },
      { name: 'a popular track', score: 2 },
    ]);
    server = await startOpenwhydServer(START_WITH_ENV_FILE);
    const { body } = await httpClient.get({ url: `${server.URL}/hot` });
    expect(body).toMatchSnapshot();
    // Note: the request above does not mutate data => no need to snapshot the state of the "tracks" table.
  });

  it("updates the score of a track when it's reposted", async () => {
    const users = [
      { id: 0, name: 'user 0', pwd: '123' },
      { id: 1, name: 'user 1', pwd: '456' },
    ];
    await db.collection('user').insertMany(users);
    server = await startOpenwhydServer(START_WITH_ENV_FILE);
    const userSession = [
      await httpClient.post({ url: `${server.URL}/api/login`, body: users[0] }),
      await httpClient.post({ url: `${server.URL}/api/login`, body: users[1] }),
    ];
    const posts = [
      // user 0 posts track A
      await httpClient.post({
        url: `${server.URL}/api/post`,
        body: { action: 'insert', eId: 'track_A' },
        cookies: userSession[0].cookies,
      }),
      // user 0 posts track B
      await httpClient.post({
        url: `${server.URL}/api/post`,
        body: { action: 'insert', eId: 'track_B' },
        cookies: userSession[0].cookies,
      }),
    ];
    // user 1 reposts track A
    await httpClient.post({
      url: `${server.URL}/api/post`,
      body: { action: 'insert', pId: posts[0].pId },
      cookies: userSession[1].cookies,
    });
    const { body } = await httpClient.get({
      url: `${server.URL}/hot?format=json`,
    });
    expect(body).toMatchSnapshot();
    // Note: the requests above mutate data => we snapshot the state of the "tracks" table.
    const tracksCollection = await db.collection('tracks').find({}).toArray();
    expect(tracksCollection).toMatchSnapshot();
  });
});
