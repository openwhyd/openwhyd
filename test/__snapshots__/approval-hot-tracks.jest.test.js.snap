// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hot Tracks (approval tests - to be replaced later by unit tests) renders ranked tracks 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head prefix=\\"og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# whydapp: http://ogp.me/ns/fb/whydapp#\\">
    <meta charset=\\"utf-8\\" />
<meta name=\\"google-site-verification\\" content=\\"mmqzgEU1bjTfJ__nW6zioi7O9vuur1SyYfW44DH6ozg\\" />
<meta name=\\"apple-itunes-app\\" content=\\"app-id=874380201\\">
<link rel=\\"image_src\\" href=\\"http://localhost:8080/images/logo-black-square-smaller.png\\"/>
<meta name=\\"description\\" content=\\"Discover and collect music gems from Youtube, Soundcloud, Deezer and more\\" />
<meta name=\\"keywords\\" content=\\"discover, music, curation, streaming, tracks, youtube, soundcloud, bandcamp, playlists, play, free\\" />
<meta name=\\"twitter:card\\" content=\\"summary\\" />
<meta name=\\"twitter:site\\" content=\\"@open_whyd\\" />
<meta property=\\"og:image\\" content=\\"http://localhost:8080/images/logo-black-square-smaller.png\\" />
<meta property=\\"og:description\\" content=\\"Discover and collect music gems from Youtube, Soundcloud, Deezer and more\\" />
<meta property=\\"fb:app_id\\" content=\\"1573219269412628\\" />
<meta property=\\"fb:admins\\" content=\\"510739408\\" />
<meta property=\\"og:type\\" content=\\"website\\" />
<meta property=\\"og:title\\" content=\\"Oops...\\" />
    <link href=\\"/favicon.ico\\" rel=\\"shortcut icon\\" type=\\"image/x-icon\\" />
    <link href=\\"/favicon.png\\" rel=\\"icon\\" type=\\"image/png\\" />
    <link href='//fonts.googleapis.com/css?family=Varela+Round' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Varela' rel='stylesheet' type='text/css'>
    <link rel=\\"search\\" type=\\"application/opensearchdescription+xml\\" title=\\"Whyd\\" href=\\"http://localhost:8080/html/opensearch.xml\\">
    <link rel=\\"chrome-webstore-item\\" href=\\"https://chrome.google.com/webstore/detail/foohaghobcolamikniehcnnijdjehfjk\\">

<script>
  window.user = {};
  window.playTrack = window.playTrack || function(){};
</script>
<script src=\\"/js/whydtr.js?1.46.1\\"></script>
<link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/css/cookieconsent2-3.0.3.min.css\\" />
<script src=\\"/js/cookieconsent2-3.0.3.min.js\\"></script>
<script>
  /* generated from https://cookieconsent.insites.com/download/ */
  window.addEventListener(\\"load\\", function(){
    window.cookieconsent.initialise({
      palette: { popup: { background: \\"#000\\" }, button: { background: \\"#f1d600\\" } }
    })
  });
</script>
    <title>Oops... - Openwhyd</title>
    <link href=\\"/css/common.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/browse.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/tipsy.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/userProfileV2.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/userPlaylistV2.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/dlgEditProfileCover.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <script src=\\"/js/jquery-1.10.2.min.js\\"></script>
    <script src=\\"/js/jquery-migrate-1.2.1.js\\"></script>
    <script type=\\"text/javascript\\" src=\\"/js/jquery.history.js\\"></script>
    <script src=\\"/js/soundmanager2-nodebug-jsmin.js\\"></script>
    <script>soundManager.setup({url: \\"/swf/\\", flashVersion: 9, onready: function() {soundManager.isReady=true;}});</script>
  </head>
  <body class=\\" visitor\\">
   <div id=\\"fb-root\\"></div>
<!--[if lt IE 8]>
<div class=\\"topWarning\\">Warning: your web browser is not supported by Openwhyd. Please upgrade to a modern browser.</div>
<![endif]-->
<div id=\\"header\\"><div class=\\"container\\">
  <div id=\\"headCenter\\">
   <a id=\\"logo\\" title=\\"Openwhyd\\" target=\\"_top\\" class=\\"homeLink\\" href=\\"/\\">
   </a>

  </div>
  <div id=\\"navbar\\">
   <a target=\\"_top\\" id=\\"tabStream\\" href=\\"/\\">Stream</a>
   <a target=\\"_top\\" id=\\"tabHot\\" href=\\"/hot\\">Hot Tracks</a>
  </div>
  <div id=\\"searchBar\\">
   <div class=\\"searchForm\\" id=\\"searchForm\\">
    <input name=\\"q\\" class=\\"q search-input\\" id=\\"q\\" type=\\"text\\" value=\\"\\" placeholder=\\"Paste a YouTube / Soundcloud / Bandcamp / Deezer URL\\" autocomplete=\\"off\\" />
   </div>
   <div class=\\"searchResults\\" id=\\"searchResults\\"></div>
  </div>
  <div id=\\"homeHeader\\">
   <h1>Openwhyd: The community of music lovers</h1>
	  <p>Discover and collect music gems from Youtube, Soundcloud, Deezer and more</p>
  </div>
  <div id=\\"logBox\\">
   <a id=\\"signin\\" href=\\"/login\\">Login</a>
   <a id=\\"signup\\" onclick=\\"login();\\">Sign up</a>
  </div>
</div></div>
<div id=\\"contentPane\\">
  <div id=\\"mainPanel\\">
<style>
  #header {
    display: none;
  }

  #contentPane {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    margin: 0 !important;
    padding: 0;
    background: url(/images/404-bg.jpg) no-repeat fixed 50% 50%;
    background-size: cover;
    font-family: 'Helvetica Neue', Helvetica, Sans-serif;
    color: #fff;
    -webkit-font-smoothing: antialiased;
  }

  #mainPanel {
    position: relative;
    width: 100%;
    height: 100%;
    margin: 0;
  }

  /* HEAD */

  .header {
    height: 80px;
    background: transparent;
    margin-bottom: 40px;
    border-bottom: 1px solid rgba(255, 255, 255, 0.2);
  }

  .header > div {
    position: relative;
    z-index: 2;
    min-width: 880px;
    width: 968px;
    margin: 0 auto;
    padding: 0;
    background: none;
  }

  .header #headCenter {
    position: absolute;
    top: 20px;
    left: 1px;
  }

  .background {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -1;

    background: rgba(0, 0, 0, 0.5);
    background: linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -webkit-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -webkit-gradient(
      linear,
      left top,
      left bottom,
      color-stop(0%, rgba(0, 0, 0, 0.75)),
      color-stop(100%, rgba(0, 0, 0, 0.2) 100%)
    );
    background: -moz-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -o-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -ms-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
  }

  .wrapper {
    width: 960px;
    margin: 120px auto;
    text-align: center;
  }

  h1 {
    margin: 0;
    padding: 0;
    font-size: 100px;
    font-weight: 500;
  }

  h2 {
    margin: -20px 0 0 0;
    padding: 0;
    font-size: 48px;
    font-weight: normal;
    color: white;
  }

  .jamming {
    font-size: 30px;
    font-weight: 300;
    text-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
    text-transform: uppercase;
  }

  a.btnPlug {
    display: block;
    width: 280px;
    height: 75px;
    line-height: 75px;
    margin: 30px auto;
    border-radius: 5px;
    color: #fff;
    font-size: 24px;
    font-weight: 500;
    text-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
    text-decoration: none;

    background: #a5c052;
    background: linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -webkit-linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -webkit-gradient(
      linear,
      left top,
      left bottom,
      color-stop(0%, #b2c967),
      color-stop(100%, #a5c052 100%)
    );
    background: -moz-linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -o-linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -ms-linear-gradient(top, #b2c967 0%, #a5c052 100%);
  }

  a.btnPlug:active {
    background: #a5c052;
  }
</style>

<div class=\\"header\\">
  <div>
    <div id=\\"headCenter\\">
      <a target=\\"_top\\" href=\\"/\\">
        <img src=\\"/images/logo-white.png\\" />
      </a>
    </div>
  </div>
</div>

<div class=\\"background\\"></div>

<div id=\\"content\\">
  <div class=\\"wrapper\\">
    <h1>404</h1>
    <h2>There is no music here</h2>
    <p><a href=\\"/\\" class=\\"btnPlug\\">Plug back in</a></p>
    <p class=\\"jamming\\">Never stop jamming</p>
  </div>
</div>

  </div>
</div>
<div id=\\"playerContainer\\">
  <div id=\\"playerContainerSub\\">
    <div
      id=\\"playerClose\\"
      onclick=\\"window.whydPlayer.toggleVideo();\\"
      title=\\"Hide\\"
    ></div>
    <div id=\\"playBtnOverlay\\" onclick=\\"window.whydPlayer.playPause();\\"></div>
  </div>
</div>
<div id=\\"whydPlayer\\">
  <div id=\\"progressTrack\\" class=\\"progressTrack\\">
    <div id=\\"progressBar\\" class=\\"progressBar\\"></div>
    <div id=\\"progressCursor\\" class=\\"progressCursor\\"></div>
  </div>
  <div class=\\"rightButtons buttons\\">
    <div
      id=\\"btnShuffle\\"
      onclick=\\"whydPlayer.toggleShuffle()\\"
      title=\\"Shuffle mode\\"
    >
      <div></div>
    </div>
    <div class=\\"volumePanel\\">
      <div class=\\"volume\\"></div>
      <div id=\\"volumeTrack\\" class=\\"progressTrack\\">
        <div class=\\"progressBar\\"></div>
        <div class=\\"progressCursor\\"></div>
      </div>
    </div>
    <div
      id=\\"btnFullscreen\\"
      onclick=\\"whydPlayer.toggleFullscreen()\\"
      title=\\"Fullscreen\\"
    >
      <div></div>
    </div>
  </div>
  <div class=\\"progressPanel buttons\\">
    <button id=\\"btnPrev\\" onclick=\\"whydPlayer.prev()\\"></button>
    <button id=\\"btnPlay\\" onclick=\\"whydPlayer.playPause()\\"></button>
    <button id=\\"btnNext\\" onclick=\\"whydPlayer.next()\\"></button>
    <button id=\\"btnRepost\\" onclick=\\"whydPlayer.repost()\\" title=\\"Add\\">
      <div></div>
    </button>
    <button id=\\"btnLike\\" onclick=\\"whydPlayer.like()\\" title=\\"Like\\">
      <div></div>
    </button>
    <button id=\\"btnComment\\" onclick=\\"whydPlayer.comment()\\" title=\\"Comment\\">
      <div></div>
    </button>
    <span id=\\"trackTitle\\">(none)</span>
    <span id=\\"progressTimer\\"></span>
  </div>
  <div id=\\"trackDragPos\\"></div>
</div>




<script>
var DEEZER_APP_ID = 190482;
var DEEZER_CHANNEL_URL = window.location.href.substr(0, window.location.href.indexOf(\\"/\\", 10)) + \\"/html/channel.html\\";
var SOUNDCLOUD_CLIENT_ID = \\"eb257e698774349c22b0b727df0238ad\\";
var YOUTUBE_API_KEY = \\"AIzaSyBzqb519R--gKZ9cXgbqE5bMM85yNTXJfo\\";
var JAMENDO_CLIENT_ID = \\"2c9a11b9\\";
</script>
    <script src=\\"/js/jquery.avgrund.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.tipsy.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/quickSearch.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.iframe-post-form.min.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.placeholder.min.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/underscore-min.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.elastic.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.mentionsInput.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/ui.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/whyd.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/playem-all.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/playem-youtube-iframe-patch.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/whydPlayer.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/dndUpload.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/WhydImgUpload.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/facebook.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
  </body>
</html>"
`;

exports[`Hot Tracks (approval tests - to be replaced later by unit tests) updates the score of a track when it's reposted 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head prefix=\\"og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# whydapp: http://ogp.me/ns/fb/whydapp#\\">
    <meta charset=\\"utf-8\\" />
<meta name=\\"google-site-verification\\" content=\\"mmqzgEU1bjTfJ__nW6zioi7O9vuur1SyYfW44DH6ozg\\" />
<meta name=\\"apple-itunes-app\\" content=\\"app-id=874380201\\">
<link rel=\\"image_src\\" href=\\"http://localhost:8080/images/logo-black-square-smaller.png\\"/>
<meta name=\\"description\\" content=\\"Discover and collect music gems from Youtube, Soundcloud, Deezer and more\\" />
<meta name=\\"keywords\\" content=\\"discover, music, curation, streaming, tracks, youtube, soundcloud, bandcamp, playlists, play, free\\" />
<meta name=\\"twitter:card\\" content=\\"summary\\" />
<meta name=\\"twitter:site\\" content=\\"@open_whyd\\" />
<meta property=\\"og:image\\" content=\\"http://localhost:8080/images/logo-black-square-smaller.png\\" />
<meta property=\\"og:description\\" content=\\"Discover and collect music gems from Youtube, Soundcloud, Deezer and more\\" />
<meta property=\\"fb:app_id\\" content=\\"1573219269412628\\" />
<meta property=\\"fb:admins\\" content=\\"510739408\\" />
<meta property=\\"og:type\\" content=\\"website\\" />
<meta property=\\"og:title\\" content=\\"Oops...\\" />
    <link href=\\"/favicon.ico\\" rel=\\"shortcut icon\\" type=\\"image/x-icon\\" />
    <link href=\\"/favicon.png\\" rel=\\"icon\\" type=\\"image/png\\" />
    <link href='//fonts.googleapis.com/css?family=Varela+Round' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Varela' rel='stylesheet' type='text/css'>
    <link rel=\\"search\\" type=\\"application/opensearchdescription+xml\\" title=\\"Whyd\\" href=\\"http://localhost:8080/html/opensearch.xml\\">
    <link rel=\\"chrome-webstore-item\\" href=\\"https://chrome.google.com/webstore/detail/foohaghobcolamikniehcnnijdjehfjk\\">

<script>
  window.user = {};
  window.playTrack = window.playTrack || function(){};
</script>
<script src=\\"/js/whydtr.js?1.46.1\\"></script>
<link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/css/cookieconsent2-3.0.3.min.css\\" />
<script src=\\"/js/cookieconsent2-3.0.3.min.js\\"></script>
<script>
  /* generated from https://cookieconsent.insites.com/download/ */
  window.addEventListener(\\"load\\", function(){
    window.cookieconsent.initialise({
      palette: { popup: { background: \\"#000\\" }, button: { background: \\"#f1d600\\" } }
    })
  });
</script>
    <title>Oops... - Openwhyd</title>
    <link href=\\"/css/common.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/browse.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/tipsy.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/userProfileV2.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/userPlaylistV2.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <link href=\\"/css/dlgEditProfileCover.css?1.46.1\\" rel=\\"stylesheet\\" type=\\"text/css\\" />
    <script src=\\"/js/jquery-1.10.2.min.js\\"></script>
    <script src=\\"/js/jquery-migrate-1.2.1.js\\"></script>
    <script type=\\"text/javascript\\" src=\\"/js/jquery.history.js\\"></script>
    <script src=\\"/js/soundmanager2-nodebug-jsmin.js\\"></script>
    <script>soundManager.setup({url: \\"/swf/\\", flashVersion: 9, onready: function() {soundManager.isReady=true;}});</script>
  </head>
  <body class=\\" visitor\\">
   <div id=\\"fb-root\\"></div>
<!--[if lt IE 8]>
<div class=\\"topWarning\\">Warning: your web browser is not supported by Openwhyd. Please upgrade to a modern browser.</div>
<![endif]-->
<div id=\\"header\\"><div class=\\"container\\">
  <div id=\\"headCenter\\">
   <a id=\\"logo\\" title=\\"Openwhyd\\" target=\\"_top\\" class=\\"homeLink\\" href=\\"/\\">
   </a>

  </div>
  <div id=\\"navbar\\">
   <a target=\\"_top\\" id=\\"tabStream\\" href=\\"/\\">Stream</a>
   <a target=\\"_top\\" id=\\"tabHot\\" href=\\"/hot\\">Hot Tracks</a>
  </div>
  <div id=\\"searchBar\\">
   <div class=\\"searchForm\\" id=\\"searchForm\\">
    <input name=\\"q\\" class=\\"q search-input\\" id=\\"q\\" type=\\"text\\" value=\\"\\" placeholder=\\"Paste a YouTube / Soundcloud / Bandcamp / Deezer URL\\" autocomplete=\\"off\\" />
   </div>
   <div class=\\"searchResults\\" id=\\"searchResults\\"></div>
  </div>
  <div id=\\"homeHeader\\">
   <h1>Openwhyd: The community of music lovers</h1>
	  <p>Discover and collect music gems from Youtube, Soundcloud, Deezer and more</p>
  </div>
  <div id=\\"logBox\\">
   <a id=\\"signin\\" href=\\"/login\\">Login</a>
   <a id=\\"signup\\" onclick=\\"login();\\">Sign up</a>
  </div>
</div></div>
<div id=\\"contentPane\\">
  <div id=\\"mainPanel\\">
<style>
  #header {
    display: none;
  }

  #contentPane {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    margin: 0 !important;
    padding: 0;
    background: url(/images/404-bg.jpg) no-repeat fixed 50% 50%;
    background-size: cover;
    font-family: 'Helvetica Neue', Helvetica, Sans-serif;
    color: #fff;
    -webkit-font-smoothing: antialiased;
  }

  #mainPanel {
    position: relative;
    width: 100%;
    height: 100%;
    margin: 0;
  }

  /* HEAD */

  .header {
    height: 80px;
    background: transparent;
    margin-bottom: 40px;
    border-bottom: 1px solid rgba(255, 255, 255, 0.2);
  }

  .header > div {
    position: relative;
    z-index: 2;
    min-width: 880px;
    width: 968px;
    margin: 0 auto;
    padding: 0;
    background: none;
  }

  .header #headCenter {
    position: absolute;
    top: 20px;
    left: 1px;
  }

  .background {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -1;

    background: rgba(0, 0, 0, 0.5);
    background: linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -webkit-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -webkit-gradient(
      linear,
      left top,
      left bottom,
      color-stop(0%, rgba(0, 0, 0, 0.75)),
      color-stop(100%, rgba(0, 0, 0, 0.2) 100%)
    );
    background: -moz-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -o-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
    background: -ms-linear-gradient(
      top,
      rgba(0, 0, 0, 0.75) 0%,
      rgba(0, 0, 0, 0.2) 100%
    );
  }

  .wrapper {
    width: 960px;
    margin: 120px auto;
    text-align: center;
  }

  h1 {
    margin: 0;
    padding: 0;
    font-size: 100px;
    font-weight: 500;
  }

  h2 {
    margin: -20px 0 0 0;
    padding: 0;
    font-size: 48px;
    font-weight: normal;
    color: white;
  }

  .jamming {
    font-size: 30px;
    font-weight: 300;
    text-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
    text-transform: uppercase;
  }

  a.btnPlug {
    display: block;
    width: 280px;
    height: 75px;
    line-height: 75px;
    margin: 30px auto;
    border-radius: 5px;
    color: #fff;
    font-size: 24px;
    font-weight: 500;
    text-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
    text-decoration: none;

    background: #a5c052;
    background: linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -webkit-linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -webkit-gradient(
      linear,
      left top,
      left bottom,
      color-stop(0%, #b2c967),
      color-stop(100%, #a5c052 100%)
    );
    background: -moz-linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -o-linear-gradient(top, #b2c967 0%, #a5c052 100%);
    background: -ms-linear-gradient(top, #b2c967 0%, #a5c052 100%);
  }

  a.btnPlug:active {
    background: #a5c052;
  }
</style>

<div class=\\"header\\">
  <div>
    <div id=\\"headCenter\\">
      <a target=\\"_top\\" href=\\"/\\">
        <img src=\\"/images/logo-white.png\\" />
      </a>
    </div>
  </div>
</div>

<div class=\\"background\\"></div>

<div id=\\"content\\">
  <div class=\\"wrapper\\">
    <h1>404</h1>
    <h2>There is no music here</h2>
    <p><a href=\\"/\\" class=\\"btnPlug\\">Plug back in</a></p>
    <p class=\\"jamming\\">Never stop jamming</p>
  </div>
</div>

  </div>
</div>
<div id=\\"playerContainer\\">
  <div id=\\"playerContainerSub\\">
    <div
      id=\\"playerClose\\"
      onclick=\\"window.whydPlayer.toggleVideo();\\"
      title=\\"Hide\\"
    ></div>
    <div id=\\"playBtnOverlay\\" onclick=\\"window.whydPlayer.playPause();\\"></div>
  </div>
</div>
<div id=\\"whydPlayer\\">
  <div id=\\"progressTrack\\" class=\\"progressTrack\\">
    <div id=\\"progressBar\\" class=\\"progressBar\\"></div>
    <div id=\\"progressCursor\\" class=\\"progressCursor\\"></div>
  </div>
  <div class=\\"rightButtons buttons\\">
    <div
      id=\\"btnShuffle\\"
      onclick=\\"whydPlayer.toggleShuffle()\\"
      title=\\"Shuffle mode\\"
    >
      <div></div>
    </div>
    <div class=\\"volumePanel\\">
      <div class=\\"volume\\"></div>
      <div id=\\"volumeTrack\\" class=\\"progressTrack\\">
        <div class=\\"progressBar\\"></div>
        <div class=\\"progressCursor\\"></div>
      </div>
    </div>
    <div
      id=\\"btnFullscreen\\"
      onclick=\\"whydPlayer.toggleFullscreen()\\"
      title=\\"Fullscreen\\"
    >
      <div></div>
    </div>
  </div>
  <div class=\\"progressPanel buttons\\">
    <button id=\\"btnPrev\\" onclick=\\"whydPlayer.prev()\\"></button>
    <button id=\\"btnPlay\\" onclick=\\"whydPlayer.playPause()\\"></button>
    <button id=\\"btnNext\\" onclick=\\"whydPlayer.next()\\"></button>
    <button id=\\"btnRepost\\" onclick=\\"whydPlayer.repost()\\" title=\\"Add\\">
      <div></div>
    </button>
    <button id=\\"btnLike\\" onclick=\\"whydPlayer.like()\\" title=\\"Like\\">
      <div></div>
    </button>
    <button id=\\"btnComment\\" onclick=\\"whydPlayer.comment()\\" title=\\"Comment\\">
      <div></div>
    </button>
    <span id=\\"trackTitle\\">(none)</span>
    <span id=\\"progressTimer\\"></span>
  </div>
  <div id=\\"trackDragPos\\"></div>
</div>




<script>
var DEEZER_APP_ID = 190482;
var DEEZER_CHANNEL_URL = window.location.href.substr(0, window.location.href.indexOf(\\"/\\", 10)) + \\"/html/channel.html\\";
var SOUNDCLOUD_CLIENT_ID = \\"eb257e698774349c22b0b727df0238ad\\";
var YOUTUBE_API_KEY = \\"AIzaSyBzqb519R--gKZ9cXgbqE5bMM85yNTXJfo\\";
var JAMENDO_CLIENT_ID = \\"2c9a11b9\\";
</script>
    <script src=\\"/js/jquery.avgrund.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.tipsy.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/quickSearch.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.iframe-post-form.min.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.placeholder.min.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/underscore-min.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.elastic.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/jquery.mentionsInput.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/ui.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/whyd.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/playem-all.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/playem-youtube-iframe-patch.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/whydPlayer.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/dndUpload.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/WhydImgUpload.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
    <script src=\\"/js/facebook.js?1.46.1\\" type=\\"text/javascript\\" charset=\\"utf-8\\"></script>
  </body>
</html>"
`;

exports[`Hot Tracks (approval tests - to be replaced later by unit tests) updates the score of a track when it's reposted 2`] = `Array []`;
